segregation/local/local_multi_diversity.py

```diff
@@ -0,0 +1,102 @@
+"""Multigroup dissimilarity index"""
+
+__author__ = "Renan X. Cortes <renanc@ucr.edu>, Sergio J. Rey <sergio.rey@ucr.edu> and Elijah Knaap <elijah.knaap@ucr.edu>"
+
+import numpy as np
+
+from .._base import MultiGroupIndex, SpatialImplicitIndex
+
+np.seterr(divide="ignore", invalid="ignore")
+
+
+def _multi_local_diversity(data, groups):
+    """
+    Calculation of Local Diversity index for each group and unit
+
+    Parameters
+    ----------
+
+    data   : a pandas DataFrame of n rows
+    
+    groups : list of strings of length k.
+             The variables names in data of the groups of interest of the analysis.
+
+    Returns
+    -------
+
+    statistics : np.array(n,k)
+                 Local Diversity values for each group and unit
+                
+    core_data  : a pandas DataFrame
```

Comment:
The PR looks good to me. But wasn't supposed to return `data` rather than `core_data`?

Reply:
its ok here because we [set the class attribute](https://github.com/pysal/segregation/blob/cf882109dc935f8da10fab16fc8c5b58715b2595/segregation/local/local_multi_diversity.py#L100) appropriately (so even though we return the variable `core_data` above, its contents get set to the class attribute named `data`)

Reply:
Cool, gotcha!